src/datomish/exec.cljc

```diff
@@ -0,0 +1,67 @@
+(ns datomish.exec
+  #?(:cljs
+     (:require-macros
+      [datomish.pair-chan :refer [go-pair <?]]
+      [cljs.core.async.macros :refer [go]]))
+  (:require
+     [datomish.sqlite :as s]
+     [datomish.sqlite-schema :as ss]
+     [datomish.query :as dq]
+     #?@(:clj
+           [[datomish.jdbc-sqlite :as j]
+            [datomish.pair-chan :refer [go-pair <?]]
+            [clojure.core.async :refer [go <!! <! >! chan close! take!]]])
+     #?@(:cljs
+           [[datomish.promise-sqlite :as j]
+            [datomish.pair-chan]
+            [cljs.core.async :as a :refer [<!! <! >! chan close! take!]]])))
+
+(defn <run
```

Comment:
I'm not convinced streaming is the right approach.  I think I'd rather see us pull into memory; streaming across channels is fraught.